src/service.rs

```diff
@@ -3,33 +3,49 @@ pub mod pb {
 }
 
 use crate::cache;
+use crate::cache::Cached::*;
 use crate::cache::{HeadEntry, ProfileEntry, SkinEntry, UuidEntry, XenosCache};
 use crate::error::XenosError;
-use crate::error::XenosError::{InvalidTextures, NoContent, NotFound, NotRetrieved, Reqwest};
+use crate::error::XenosError::{InvalidTextures, NotFound, NotRetrieved};
 use crate::mojang::{MojangApi, UsernameResolved};
 use crate::service::pb::{
     HeadRequest, HeadResponse, ProfileProperty, ProfileRequest, ProfileResponse, SkinRequest,
     SkinResponse,
 };
-use crate::util::{get_epoch_seconds, has_elapsed};
+use crate::util::get_epoch_seconds;
 use image::{imageops, ColorType, GenericImageView, ImageOutputFormat};
 use lazy_static::lazy_static;
 use pb::profile_server::Profile;
 use pb::{UuidRequest, UuidResponse, UuidResult};
+use prometheus::{register_histogram_vec, register_int_counter_vec, HistogramVec, IntCounterVec};
 use regex::Regex;
-use reqwest::StatusCode;
 use std::collections::HashMap;
 use std::io::Cursor;
 use tokio::sync::Mutex;
 use tonic::{Request, Response, Status};
 use uuid::Uuid;
 
-const CACHE_TIME: u64 = 5 * 60;
-
 lazy_static! {
     static ref USERNAME_REGEX: Regex = Regex::new("^[a-zA-Z0-9_]{2,16}$").unwrap();
 }
 
+// TODO integrate metrics
+lazy_static! {
+    pub static ref PROFILE_REQ_TOTAL: IntCounterVec = register_int_counter_vec!(
+        "profile_requests_total",
+        "Total number of requests to the profile grpc service.",
+        &["request_type", "status", "cached"],
```

Comment:
For this and the histogram: Let's use `cache_result` instead of `cached` and have the values `hit`, `expired` and `miss`.